### DISCUSSION TITLE
Scrolling a list using <Transition />

### DISCUSSION THREAD

**Discussion by cleoold** [pmndrs/react-spring#928] [Support]
## 🤓 Question

Hello. I have a functional component that fetches an array of data and displays them in an `<ul />` which can look like:

```js
function MyLiItem({ entry, style }) {
    return <li style={style}>{entry.text}</li>
}

function Tile({ style }) {
    const { success, data } = useDataFetching(.....);
    const dataLen = length of fetched array
    const { currentIndeces } = state, will alternate among say [0,1,2], [1,2,3],[2,3,0] at an interval;
    return (
           <ul>
                <Transition
                    items={currentIndeces}
                    from={{ opacity: 0, transform: 'translateY(15%)'}}
                    enter={{ opacity: 1, transform: 'translateY(0)' }}
                    leave={{ opacity: 0, transform: 'translateY(-25%)', position: 'absolute' }}>
                    {i => style => 
                        <MyLiItem entry={data[i]} style={style} key={...something}/>}
                </Transition>
         </ul>
    )
}
```
If I display 3 items at a time, when it is time for flipping data, only the first one (which gets removed) and the last (which is the new one) are animated, the ones in the middle will jump right to the top without transitions. How can one animate the middle one so that it smoothly moves from the 2nd line to the 1st line. Can you by any way use multiple animated tags at once?

**Comment by karan1919** [comment]
hey @cleoold , you can use the 'useChain' hook to achieve this. here is a sample code https://codesandbox.io/s/vibrant-franklin-yxlfm
